libsolidity/ast/Types.cpp

```diff
@@ -2252,6 +2266,30 @@ u256 StructType::storageSize() const
 	return max<u256>(1, members(nullptr).storageSize());
 }
 
+bool StructType::containsNestedMapping() const
+{
+	if (!m_struct.annotation().containsNestedMapping.has_value())
+	{
+		std::set<TypePointer> visitedTypes = util::BreadthFirstSearch<TypePointer>{{this}}.run(
+			[](Type const* _type, auto&& _addChild)
+			{
+				if (auto _structType = dynamic_cast<StructType const*>(_type))
+					for (auto member: _structType->structDefinition().members())
+						_addChild(member->type());
+				else if (auto _arrayType = dynamic_cast<ArrayType const*>(_type))
+					_addChild(_arrayType->finalBaseType(false));
```

Comment:
Is it safe to work with pointers here? You could run into a recursion that is not detected because we create new type pointers each time.

Reply:
Good point. I wrote a wrapper for `Type const*` that now compares types based on `Type::richIdentifier()`.  What do you think about it?